dom/src/main/java/org/apache/james/mime4j/message/MessageBuilder.java

```diff
@@ -75,15 +75,68 @@
     private BodyFactory bodyFactory;
     private boolean flatMode;
     private boolean rawContent;
+    private final boolean ensureDateNotNull;
```

Comment:
I agree with Luc arguments, they are strong and sound.
The only downside of the current solution IMO is the fact that you have a contextual builder : depending on the way you build the object, it has different behavior, it can be hard to track when bug occurs.
I would in fact create a new type.

And to answer API stability point, which is important too : it's why deprecation is for, you don't break anything, you just say you won't fix that code and eventually remove it in the future (or not, see jdk deprecation ... )